dashboard/app/controllers/levels_controller.rb

```diff
@@ -59,13 +59,55 @@ class LevelsController < ApplicationController
   # GET /levels
   # GET /levels.json
   def index
+    # Define search filter fields
+    @search_fields = [
+      {
+        name: :name,
+        description: 'Filter by name:',
+        type: 'text'
+      },
+      {
+        name: :level_type,
+        description: 'By type:',
+        type: 'select',
+        options: [
+          ['All types', ''],
+          *LEVEL_CLASSES.map {|x| [x.name, x.name]}.sort_by {|a| a[0]}
+        ]
+      },
+      {
+        name: :script_id,
+        description: 'By script:',
+        type: 'select',
+        options: [
+          ['All scripts', ''],
+          *Script.valid_scripts(current_user).pluck(:name, :id).sort_by {|a| a[0]}
```

Comment:
nit: could use `.order(:name)` before `.pluck` to simplify the sorting syntax, here and below